jdaviz/configs/mosviz/helper.py

```diff
@@ -528,19 +579,180 @@ def load_images(self, data_obj, data_labels=None, share_image=0):
         super().load_data(data_obj, parser_reference="mosviz-image-parser",
                           data_labels=data_labels, share_image=share_image)
 
-    def add_column(self, data, column_name=None):
+    def get_column_names(self, visible=None):
+        """
+        List the names of the columns in the table.
+
+        Parameters
+        ----------
+        visible: bool or None
+            If None (default): will show all available column names.
+            If True: will only show columns names currently shown in the table.
+            If False: will only show column names currently not shown in the table.
+        """
+        if visible is None:
+            return [c.label for c in self.app.data_collection['MOS Table'].components]
+        elif visible is True:
+            return [h['value'] for h in self.app.get_viewer('table-viewer').widget_table.headers]
+        elif visible is False:
+            return [cn for cn in self.get_column_names() if cn not in self.get_column_names(True)]
+        else:
+            raise ValueError("visible must be one of None, True, or False.")
+
+    def set_visible_columns(self, column_names):
+        """
+        Set the columns to be visible in the table.
+
+        Parameters
+        ----------
+        column_names: list
+            list of columns to be visible in the table.
+        """
+        if not isinstance(column_names, list):
+            raise TypeError("column_names must be of type list")
+        avail_names = self.get_column_names()
+        if not np.all([c in avail_names for c in column_names]):
+            raise ValueError("not all entries of column_names are valid")
+        is_sortable = ['Redshift']
+        headers = [{'text': cn, 'value': cn, 'sortable': cn in is_sortable} for cn in column_names]
+        wt = self.app.get_viewer('table-viewer').widget_table
+        wt.set_state({'headers': headers})
+        wt.send_state()
+
+    def hide_column(self, column_name):
```

Comment:
I added a check to raise a ValueError if passing a non-existing column name to `hide_column`.  Similarly to `show_column`, it will silently ignore cases where you pass a column name that is already of the desired new state (passing an already hidden column to `hide_column` will not complain).